src/socket/dhcpv4.rs

```diff
@@ -104,32 +99,71 @@ enum ClientState {
     Renewing(RenewState),
 }
 
+/// Timeout and retry configuration.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct RetryConfig {
+    pub discover_timeout: Duration,
+    /// The REQUEST timeout doubles every 2 tries.
+    pub initial_request_timeout: Duration,
+    pub request_retries: u16,
+    pub min_renew_timeout: Duration,
+}
+
+impl RetryConfig {
+    pub const DEFAULT: Self = Self {
+        discover_timeout: Duration::from_secs(10),
+        initial_request_timeout: Duration::from_secs(5),
+        request_retries: 5,
+        min_renew_timeout: Duration::from_secs(60),
+    };
+}
+
+impl Default for RetryConfig {
+    fn default() -> Self {
+        Self::DEFAULT
+    }
+}
+
 /// Return value for the `Dhcpv4Socket::poll` function
 #[derive(Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
-pub enum Event {
+pub enum Event<'a> {
     /// Configuration has been lost (for example, the lease has expired)
     Deconfigured,
     /// Configuration has been newly acquired, or modified.
-    Configured(Config),
+    Configured(Config<'a>),
 }
 
 #[derive(Debug)]
-pub struct Socket {
+pub struct Socket<'a> {
     /// State of the DHCP client.
     state: ClientState,
-    /// Set to true on config/state change, cleared back to false by the `config` function.
+    /// Set to true on config/state change, cleared back to false by the
+    /// `config` function.
     config_changed: bool,
     /// xid of the last sent message.
     transaction_id: u32,
 
-    /// Max lease duration. If set, it sets a maximum cap to the server-provided lease duration.
-    /// Useful to react faster to IP configuration changes and to test whether renews work correctly.
+    /// Max lease duration. If set, it sets a maximum cap to the server-provided
+    /// lease duration. Useful to react faster to IP configuration changes
+    /// and to test whether renews work correctly.
     max_lease_duration: Option<Duration>,
 
+    retry_config: &'a RetryConfig,
```

Comment:
this can be owned instead of a reference

Reply:
Done.